src/gateway_config_miner.erl

```diff
@@ -0,0 +1,98 @@
+-module(gateway_config_miner).
+
+-behavior(gen_server).
+
+-include("gateway_config.hrl").
+-define(SERVICE, 'helium.local.api').
+
+%% gen_server
+-export([
+    init/1,
+    handle_call/3,
+    handle_cast/2,
+    handle_info/2
+]).
+
+%% api
+-export([
+    status/0,
+    pubkey/0,
+    add_gateway/3
+]).
+
+-record(state, {
+    connection = undefined :: grpc_client:connection() | undefined
+}).
+
+-spec status() -> [tuple()].
+status() ->
+    gen_server:call(?MODULE, status).
+
+-spec pubkey() -> {PubKey :: binary(), OnboardingKey :: binary()}.
+pubkey() ->
+    gen_server:call(?MODULE, pubkey).
+
+-spec add_gateway(
+    Owner :: libp2p_crypto:pubkey_bin(),
+    Payer :: libp2p_crypto:pubkey_bin(),
+    Mode :: full | dataonly | light
+) -> {ok, binary()} | {error, term()}.
+add_gateway(Owner, Payer, Mode) ->
+    gen_server:call(?MODULE, {add_gateway, [Owner, Payer, Mode]}).
+
+init(_) ->
+    erlang:register(?MODULE, self()),
+    {ok, #state{}}.
+
+handle_call(Call, From, State = #state{connection = undefined}) ->
+    case grpc_client:connect(tcp, "localhost", 4467) of
```

Comment:
```suggestion
    GrpcPort = application:get_env(gateway_confid, grpc_port, 4467),
    case grpc_client:connect(tcp, "localhost", GrpcPort) of
```
currently the rust gateway's grpc api port is at 4468 on embedded miners so as not to conflict with the existing 4467 in use by the legacy miner JSON RPC port. not sure how we want to handle this in LGW-only land or even if we'll still be shipping gateway-config alongside the LGW-only miners but this should probably be configurable to handle either port